util.sh

```diff
@@ -0,0 +1,147 @@
+#!/bin/bash
+
+# exit on any error
+set -e
+
+readonly ROOT=$(dirname "${BASH_SOURCE}")
+source "${ROOT}/${KUBE_CONFIG_FILE:-"config-default.sh"}"
+source "${ROOT}/openrc-default.sh"
+
+DEFAULT_KUBECONFIG=~/.kube/config
+
+# Verify prereqs on host machine
+function verify-prereqs() {
+ # Check the OpenStack command-line clients
+ for client in heat nova kubectl;
+ do
+  if which $client >/dev/null 2>&1; then
+    echo "$client client installed"
+  else
+    echo "$client client does not exist"
+    echo "Please install $client client, and retry."
+    exit 1
+  fi
+ done
+}
+
+# Instantiate a kubernetes cluster
+#
+# Assumed vars:
+#   KUBERNETES_PROVIDER
+function kube-up() {
+    echo "kube-up for provider $KUBERNETES_PROVIDER"
+    create-stack
+}
+
+# Periodically checks if cluster is created
+#
+# Assumed vars:
+#   STACK_CREATE_TIMEOUT
+#   STACK_NAME
+function validate-cluster() {
+
+  while (( --$STACK_CREATE_TIMEOUT >= 0)) ;do
+     local status=$(heat stack-show "${STACK_NAME}" | awk '$2=="stack_status" {print $4}')
+     if [[ $status ]]; then
+        echo "Cluster status ${status}"
+        if [ $status = "CREATE_COMPLETE" ]; then
+          configure-kubectl
+          break
+        elif [ $status = "CREATE_FAILED" ]; then
+          echo "Cluster not created. Please check stack logs to find the problem"
+          break
+        fi
+     else
+       echo "Cluster not created. Please verify if process started correctly"
+       break
+     fi
+     sleep 30
+  done
```

Comment:
Why sleep 30?  
I think it is better to use SECONDS variable
http://linux.die.net/man/1/bash

in first of script, set `SECONDS=0`  and

```
while (( $SECONDS < ${STACK_CREATE_TIMEOUT}*60 )) ;do
   ----
```


Other part is LGTM